x-pack/plugins/uptime/server/lib/alerts/status_check.ts

```diff
@@ -287,26 +315,67 @@ export const statusCheckAlertFactory: UptimeAlertTypeFactory<ActionGroupIds> = (
       };
 
       let downMonitorsByLocation: GetMonitorStatusResult[] = [];
+      let upMonitorsByLocation: GetMonitorStatusResult[] = [];
 
       // if oldVersionTimeRange present means it's 7.7 format and
       // after that shouldCheckStatus should be explicitly false
       if (!(!oldVersionTimeRange && shouldCheckStatus === false)) {
-        downMonitorsByLocation = await libs.requests.getMonitorStatus({
+        const queryConfig = {
           uptimeEsClient,
           timerange,
           numTimes,
           locations: [],
           filters: filterString,
-        });
+        };
+        downMonitorsByLocation = (await libs.requests.getMonitorStatus(queryConfig)) || [];
+        upMonitorsByLocation =
+          (await libs.requests.getMonitorStatus({
```

Comment:
Do we need to call get up monitors every time? or only when we find no down monitors and we had down monitors in previous execution?

Reply:
Thanks for the reminder! We talked about this but I totally forgot to make this optimization after focusing on the bug we found. I'll get right to it.